### DISCUSSION TITLE
Suddenly, header ‘Prefer: params=single-object’ is required.

### DISCUSSION THREAD

**Discussion by gettes** [PostgREST/postgrest#2968] [Q&A] [answered]
I have developed an rpc with postgrest.  I pass it a single json parameter and It was working great.  And then, all of a sudden, I now need to include the header via curl -H ‘Prefer: params=single-object’.  How do I get this working again without this header?
I am using postgrest 11.2 on postgres 14.9.

Please help!

**Comment by steve-chavez** [comment]
Share the steps to reproduce the issue: minimal SQL definition plus how you make the request to PostgREST and the response body (curl snippet).

**Reply by steve-chavez** [reply]
Though it sounds like you changed your function definition and named your function parameter, see https://postgrest.org/en/stable/references/api/stored_procedures.html#functions-with-a-single-unnamed-parameter

**Reply by gettes** [reply, marked as answer]
Yeah, I confused myself with variations on the function call and needed to restart postgrest to have it pick up the schema changes.  Sorry for bothering you and THANK YOU so much for the prompt reply.

**Reply by steve-chavez** [reply]
No problem, maybe try https://postgrest.org/en/stable/references/schema_cache.html#automatic-schema-cache-reloading to not run into those issues
